src/fidesops/db/base_class.py

```diff
@@ -23,16 +24,16 @@
 
 def get_key_from_data(data: Dict[str, Any], cls_name: str) -> str:
     """
-    Extracts key from data and slugifies. If no key, uses a slugified name.
+    Extracts key from data, validates. If no key, uses a slugified name.
 
     Will be used as the URL slug on applicable classes.
     """
-    key = slugify(data.get("key")) if data.get("key") else None
+    key = FidesKey.validate(data.get("key")) if data.get("key") else None
```

Comment:
This error needs to be handled in our API endpoints - if a user submits a key with a dash for example, they will see an unhandled 500 error which doesn't help them correct their request.  


Multiple options here, we could catch the FidesValidationError and reraise a ValidationError in the endpoints, or we could add this check to the schema and use a[ Pydantic validator ](https://pydantic-docs.helpmanual.io/usage/validators/) on key (only if the user supplied it, and raise a ValueError where applicable - here's an example of a validator. https://github.com/ethyca/fidesops/blob/main/src/fidesops/schemas/storage/storage.py#L161.  We could even make a key mixin to use across a lot of the schemas:

Untested, something like:
```python
class FidesKeyMixin(BaseModel):

    key: Optional[FidesKey]

     @validator('key')
     def validate_key(cls, value):
           if value is None:
                  return value 
           try:
               FidesKey.validate(value)
           except FidesValidationError as exc:
               raise ValueError(exc)
           return value
```

Reply:
I like the idea of a key mixin, because then we can re-use the same validator across multiple schemas that use a `FidesKey` as `key`. Plus we get the benefit of bundling up the `FidesValidationError` as a `ValueError` so that our api endpoints don't get too overloaded with a bunch of custom errors.